src/DistributedLock.Redis/Primitives/RedisSemaphorePrimitive.cs

```diff
@@ -78,4 +78,30 @@ public Task ReleaseAsync(IDatabaseAsync database, bool fireAndForget) =>
     public Task<bool> TryExtendAsync(IDatabaseAsync database) => ExtendScript.ExecuteAsync(database, this).AsBooleanTask();
 
     public bool IsConnected(IDatabase database) => database.IsConnected(this._key, CommandFlags.DemandMaster);
+
+    /// <summary>
+    /// Get Current Count
+    /// </summary>
+    /// <param name="database"></param>
+    /// <returns></returns>
+    internal async ValueTask<int> GetCurrentCountAsync(IDatabaseAsync database)
+    {
+        const string script = @"return redis.call('zcard', KEYS[1])";
```

Comment:
Using a script here adds unnecessary overhead since we're just executing a single command. Also, we want to support sync-via-async execution.

Let's just do `database.SortedSetLengthAsync()` with `CommandFlags.DemandMaster`.

To support sync-via-async execution, we should actually do:

```
var length = SyncViaAsync.IsSynchronous
    ? database.SortedSetLength(...)
    :  await database.SortedSetLengthAsync(...).ConfigureAwait(false);
```